packages/atomic/src/components/commerce/product-template-components/product-utils.ts

```diff
@@ -12,3 +13,37 @@ export function getStringValueFromProductOrNull(
 
   return value;
 }
+
+export function buildStringTemplateFromProduct(
+  template: string,
+  product: Product,
+  bindings: CommerceBindings
+) {
+  return template.replace(/\${(.*?)}/g, (value: string) => {
+    const key = value.substring(2, value.length - 1);
+    let newValue = readFromObject(product, key);
+    if (!newValue) {
+      newValue = readFromObject(window, key);
+    }
+
+    if (!newValue) {
+      bindings.engine.logger.warn(
+        `${key} used in the href template is undefined for this product: ${product.permanentid}`
+      );
+      return '';
+    }
+
+    return newValue;
+  });
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function readFromObject(object: any, key: string): string {
+  const firstPeriodIndex = key.indexOf('.');
+  if (object && firstPeriodIndex !== -1) {
+    const newKey = key.substring(firstPeriodIndex + 1);
+    key = key.substring(0, firstPeriodIndex);
+    return readFromObject(object[key], newKey);
+  }
+  return object ? object[key] : undefined;
+}
```

Comment:
This seem complicated with recursion. Can't you do something like this ? 

```
  const keys = key.split('.');
  let currentObject = object;

  for (const keyPart of keys) {
    if (currentObject[keyPart] === undefined) {
      return undefined;
    }
    currentObject = currentObject[keyPart];
  }

  return currentObject;
```

Reply:
This is taken as is from result-utils.ts, but yes I can avoid the recursion